File: frontend/src/app/models/question.ts
import { Form } from './form';
import { OptionList } from './optionList' ;
import {User} from "./user";
import {OptionValue} from "./optionValue";
import {Answer} from "./answer";
export enum QuestionType {
    Short,
    Long,
    Date,
    Email,
    Integer,
    Check,  // le seul qui permette à l'utilisateur de choisir plusieurs valeurs.
    Combo,
    Radio
}
export class Question {
    
    id: number;
    idx: number;
    formId: number;
    form: Form; 
    title: string;
    description?: string;
    questionType: QuestionType;
    required: boolean;
    optionListId?: number;
    optionList?: OptionList;
    answersList?: Answer[];

    constructor(data: any) {
        this.id = data.id;
        this.idx = data.idx;
        this.formId = data.formId;
        this.form = data.form;
        this.title = data.title;
        this.description = data.description;
        this.questionType = data.questionType;
        this.required = data.required;
        this.optionListId = data.optionListId;
        this.optionList = data.optionList ? new OptionList(data.optionList) : undefined;
        this.answersList = data.answersList ? data.answersList.map((a: any) => new Answer(a)) : undefined;
    }

    toString(): string {
        return `Question {
        id: ${this.id},
        idx: ${this.idx},
        form: ${this.form ? this.form.toString() : 'undefined'},
        title: "${this.title}",
        description: "${this.description || ''}",
        questionType: ${QuestionType[this.questionType]},
        required: ${this.required},
        optionList: ${this.optionList ? this.optionList.toString() : 'undefined'}
        }`;
    }
}

File: backend/Models/Question/Question.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace prid_2425_a01.Models.Question;

public enum QuestionType {
    Short,  
    Long,   
    Date,   
    Email,  
    Integer,
    Check,  // le seul qui permette à l'utilisateur de choisir plusieurs valeurs.
    Combo,  
    Radio  
}

public class Question {

    [Key]
    public int Id { get; set; }

    [Required]
    public int FormId { get; set; }

    [Required]
    public int Idx { get; set; }

    [Required]
    public string Title { get; set; } = null!;

    public string? Description { get; set; }

    [Required]
    public QuestionType QuestionType { get; set; } 

    [Required]
    public bool Required { get; set; }

    public int? OptionListId { get; set; }
    

    [ForeignKey(nameof(FormId))]
    public Form.Form Form { get; set; } = null!;

    [ForeignKey(nameof(OptionListId))]
    public OptionList.OptionList? OptionList { get; set; }



    public ICollection<Answer.Answer> ListAnswers { get; set; } = new HashSet<Answer.Answer>();

    public override string ToString() {
        return $"Question {{ " +
            $"Id = {Id}, " +
            $"FormId = {FormId}, " +
            $"Idx = {Idx}, " +
            $"Title = \"{Title}\", " +
            $"Description = \"{Description}\", " +
            $"QuestionType = {QuestionType}, " +
            $"Required = {Required}, " +
            $"OptionListId = {OptionListId}" +
            $" }}";
    }
}
